packages/common/src/otel.ts

```diff
@@ -0,0 +1,49 @@
+import { isSpanContextValid, Span, SpanContext } from '@opentelemetry/api';
+import { defaultDataConverter } from './converter/data-converter';
+import { Headers } from './interceptors';
+
+export const TRACE_HEADER = 'Otel-Trace-Context';
+/** As in workflow run id */
+export const RUN_ID_ATTR_KEY = 'run_id';
+/** For a workflow or activity task */
+export const TASK_TOKEN_ATTR_KEY = 'task_token';
+/** Number of jobs in a workflow activation */
+export const NUM_JOBS_ATTR_KEY = 'num_jobs';
+
+/**
+ * If found, return a span context deserialized from the provided headers
+ */
+export async function extractSpanContextFromHeaders(headers: Headers): Promise<SpanContext | undefined> {
+  const encodedSpanContext = headers[TRACE_HEADER];
+  if (encodedSpanContext === undefined) {
+    return undefined;
+  }
+  const decoded: SpanContext = await defaultDataConverter.fromPayload(encodedSpanContext);
+  if (isSpanContextValid(decoded)) {
+    return decoded;
+  }
+  return undefined;
+}
+
+/**
+ * Given a span context & headers, return new headers with the span context inserted
+ */
+export async function insertSpanContextIntoHeaders(context: SpanContext, headers: Headers): Promise<Headers> {
```

Comment:
The name insert  is inaccurate since there's no mutation, maybe rename to something like `headersWithSpanContext`?